hledger/Hledger/Cli/Commands/Run.hs

```diff
@@ -108,5 +115,46 @@ runREPL findBuiltinCommand j = do
       Just "quit" -> return ()
       Just "exit" -> return ()
       Just input -> do
-        liftIO $ runCommand findBuiltinCommand j $ parseCommand input
+        liftIO $ runCommand findBuiltinCommand $ parseCommand input
         loop
+
+{-# NOINLINE journalCache #-}
+journalCache :: MVar (Map.Map [String] Journal)
+journalCache = unsafePerformIO $ newMVar Map.empty
+
+-- | Key used to cache the journal given in the arguments to 'run'.
+defaultJournalKey :: [String]
+defaultJournalKey = ["journal specified in args of run"]
+
+hasStdin :: [String] -> Bool
+hasStdin fps = "-" `elem` fps
```

Comment:
splitReaderPrefix should be used here in case it's a prefixed path like `csv:-`.

Reply:
This is fixed in later commits. Do you want me to squash everything in a single commit, perhaps?